neo4j-jdbc-http/src/main/java/org/neo4j/jdbc/http/driver/CypherExecutor.java

```diff
@@ -377,6 +386,10 @@ public Integer getOpenTransactionId() {
 		return getTransactionId(this.currentTransactionUrl);
 	}
 
+	private String transactionPath(String databaseName) {
+		return String.format(DB_DATA_TRANSACTION_TEMPLATE, databaseName);
```

Comment:
it appears there is no need for escaping, as database names can only be made of: `"simple ascii characters, numbers, dots and dashes"` and [these characters are not URI special characters](https://tools.ietf.org/html/rfc3986#section-2.3).